usaspending_api/awards/management/commands/loadcontracts.py

```diff
@@ -26,25 +27,227 @@ def handle(self, *args, **options):
+            "major_program": lambda row: self.parse_first_character(row['majorprogramcode']),
+            "program_acronym": lambda row: self.parse_first_character(row['programacronym']),
+            "referenced_idv_modification_number": lambda row: self.parse_first_character(row['idvmodificationnumber']),
+            "transaction_number": lambda row: self.parse_first_character(row['transactionnumber']),
+            "solicitation_identifier": lambda row: self.parse_first_character(row['solicitationid']),
             "submission": subattr
         }
 
         loader = ThreadedDataLoader(Procurement, field_map=field_map, value_map=value_map)
         loader.load_from_file(options['file'][0])
 
+    def evaluate_contract_award_type(self, row):
+        first_element = row['contractactiontype'].split(' ')[0].split(':')[0]
+        if len(first_element) == 1:
+            return first_element
+        else:
+            cat = row['contractactiontype'].lower()
+            # Not using DAIMS enumeration . . .
+            if 'bpa' in cat:
+                return 'A'
+            if 'purchase' in cat:
+                return 'B'
+            elif 'delivery' in cat:
+                return 'C'
+            elif 'definitive' in cat:
+                return 'D'
+            else:
+                return None
+
+    def get_agency(self, row):
+        agency = Agency.objects.filter(subtier_code=self.get_agency_code(row['maj_agency_cat'])).first()
+        if not agency:
+            print("Missing agency: " + row['maj_agency_cat'])
+        return agency
+
+    def create_or_get_recipient(self, row):
+        # First, create the locations
+        location_dict = {}
+        pop_dict = {}
+
+        loc_country_code = row["vendorcountrycode"].split(":")[0]
+
+        loc_country_code = RefCountryCode.objects.filter(Q(**{"country_code": loc_country_code}) | Q(**{"country_name__iexact": loc_country_code})).first()
+        if not loc_country_code:
+            # We don't have an exact match on the name or the code, so we need to chain filter on the name
+            query_set = RefCountryCode.objects
+            for word in row["vendorcountrycode"].split(":")[0].split(" "):
+                query_set = query_set.filter(country_name__icontains=word)
+            loc_country_code = query_set.first()
+
+        if loc_country_code.country_code == "USA":
+            location_dict = {
+                "location_country_code": loc_country_code,
+                "location_zip5": row["zipcode"][:5],
+                "location_zip_last4": row["zipcode"].replace("-", "")[5:],
+                "location_address_line1": row["streetaddress"],
+                "location_address_line2": row["streetaddress2"],
+                "location_address_line3": row["streetaddress3"],
+                "location_state_code": row["state"],
+                "location_city_name": row["city"],
```

Comment:
Thanks for the clarification. I'll add a story for the city and county codes, in case we want to circle back and update the model. I think the answer to #2 depends on how we handle #1. Agreed that it makes sense for locations to be unique via a combination of their parts. I raised the question because it doesn't look like we're using all the parts (not using the codes especially threw me off...like, what if a city name changes but the code is still the same...didn't realize we weren't getting those yet)